packages/medusa/src/services/sales-channel.ts

```diff
@@ -81,11 +81,33 @@ class SalesChannelService extends TransactionBaseService<SalesChannelService> {
     })
   }
 
+  /**
+   * Lists sales channels based on the provided parameters and includes the count of
+   * sales channels that match the query.
+   * @return an array containing the sales channels as
+   *   the first element and the total count of sales channels that matches the query
+   *   as the second element.
+   */
   async listAndCount(
-    selector: QuerySelector<any> = {},
-    config: FindConfig<any> = { relations: [], skip: 0, take: 10 }
+    selector: ListSalesChannelInput,
```

Comment:
Can we use `QuerySelector<SalesChannel>` here since it's also the type passed to the prepareListQuery?